### PR TITLE
Issue194 ERA5 and TOPAZ forcings

### LINKED ISSUE #194: Implement simplistic ERA5 and TOPAZ forcing implementations
In order to run models that are reasonably representative of the real world, real-world forcings need to be used.
For a certain grid (Arctic only, fixed resolution) implement forcing ingestion functions that work only with forcing data that has been preprocessed to match the spatial extent and resolution of the model. Time interpolation is still required.

### PR DESCRIPTION
**Pull request by timspainNERSC**
In order to run models that are reasonably representative of the real world, real-world forcings need to be used.
For a certain grid (Arctic only, fixed resolution) implement forcing ingestion functions that work only with forcing data that has been preprocessed to match the spatial extent and resolution of the model. Time interpolation is still required.

To test the functionality, one of the added changes is a year-long run of the model using forcings from ERA5 and TOPAZ. Running this is as simple as running the script `run/run_topaz128x128.sh` – if you have the necessary forcings files. Since these are ~7 GB, they are not included in the repository and need to be generated. The generation script is `run/era5_topaz4_maker.py`. This requires the ERA5 and TOPAZ4 data files, which can be found on the SASIP data repository hosted at `univ-grenoble-alpes.fr`. The necessary files are:

- `ERA5_*_y2010.nc` in `/summer/sasip/model-forcings/atmo_forcing/ERA5_Arctic`
- `TP4DAILY_2010*.nc` in `/summer/sasip/model-forcings/ocean_forcing/TOPAZ4_Arctic`

The generation script is then run in a directory containing these files with start and stop time parameters. Since the forcing files should cover all of 2010, the full command is
```
./era5_topaz4_maker.py --file init_topaz128x128.nc --start 2010-01-01 --stop 2011-01-01
```

This will generate the files `ERA5_2010-01-01_2011-01-01.nc` and `TOPAZ4_2010-01-01_2011-01-01.nc`. These file, along with the restart file `init_topaz128x128.nc`, which is included in the repository, are all that is necessary to run the example model.

To determine if the model functioned correctly, open the output file `diagnostic.nc`. This will contain output data from the model sampled once per day. The data in the field `hice_true` should expand during the first part of the year, as the thermodynamics causes the sea ice to thicken. Then over the summer, the sea ice should retreat before reforming as the end of the year approaches.

Closes #194

### CODE REVIEW

**Comment by timspainNERSC** [comment]
Since the year-long model runs successfully and there are no conflicts, I think this PR is now ready for review.

File: core/src/modules/ConfigOutput.cpp [hunk 1]
```diff
@@ -51,35 +66,93 @@ void ConfigOutput::configure()
...
+    if ((everyTS && meta.time() >= lastOutput)
+        || (std::fmod((meta.time() - lastOutput).seconds(), outputPeriod.seconds()) == 0.)) {
+        Logged::info("ConfigOutput: Outputting " + std::to_string(state.data.size()) + " fields to "
+            + currentFileName + " at " + meta.time().format() + "\n");
+        StructureFactory::fileFromState(state, meta, currentFileName, false);
         lastOutput = meta.time();
     }
 }
+
+std::string concatenateFields(const std::set<std::string>& strSet)
```
Comment (dorchard) [inline core/src/modules/ConfigOutput.cpp:139]: This will leave a dangling , comma in the output. Is that okay? (Probably it is but just wanted to check!).

**Comment by mondus** [comment]
Kacper has requested that this be rebased as some of the changes have already been merged to dev. Kacper happy to have a go at rebasing this himself once the MPI work is completed.

**Comment by mondus** [comment]
@timspainNERSC Could you add the description to how to run the year long model to the PR description. E.g Where are the input files and configurations. How do we interpret the results to ensure that this has produced the correct answer (even without the full dynamics).

**Review by einola — APPROVED** [review]

**Pull request merged**